src/Symfony/Component/Yaml/Inline.php

```diff
@@ -617,21 +617,29 @@ private static function evaluateScalar(string $scalar, int $flags, array $refere
             // Optimize for returning strings.
             // no break
             case '+' === $scalar[0] || '-' === $scalar[0] || '.' === $scalar[0] || is_numeric($scalar[0]):
+                if (Parser::preg_match('{^[+-]?[0-9][0-9_]*$}', $scalar)) {
+                    $scalar = str_replace('_', '', (string) $scalar);
+                }
+
                 switch (true) {
-                    case Parser::preg_match('{^[+-]?[0-9][0-9_]*$}', $scalar):
-                        $scalar = str_replace('_', '', (string) $scalar);
-                        // omitting the break / return as integers are handled in the next case
-                        // no break
                     case ctype_digit($scalar):
                         $raw = $scalar;
                         $cast = (int) $scalar;
 
+                        if ('0' === $scalar[0] && !Parser::preg_match('/^[0-7]*$/', $scalar)) {
+                            @trigger_error(sprintf('Support for parsing the invalid octal number %s is deprecated since Symfony 5.1.', $scalar), E_USER_DEPRECATED);
```

Comment:
should we mention it will be parsed as a string in 6+ ?

Reply:
At this point, to avoid the deprecation notice in PHP 7.4, how about removing the invalid characters before passing the scalar to octdec()?

Reply:
@cebe I moved the PHP 7.4 compat fix to #36690